src/stories/Resizeable/Resizeable.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+import DataTable from '../../lib/DataTable';
+import {YCLOUD_THEME} from '../../lib/constants';
+import type {Column, DataTableProps} from '../../lib/types';
+import {updateColumnsWidth, useTableResize} from '../../lib/useTableResize';
+import {cn} from '../../lib/util';
+
+import './Resizeable.scss';
+
+const b = cn('resizeable-datable-example');
+
+const data = Array(5)
+    .join('|')
+    .split('|')
+    .map((_empty, index) => {
+        return {
+            id: index + 1,
+            number: index + 123456789101112,
+            col1: index % 23,
+            col2: index % 13,
+            string: `some_very_long_line_${index + 1}`,
+            complex: {
+                value: index + 5,
+            },
+            bar: Math.pow(index * 20 - 250, 2) / 500,
+            something1: index + 123456789101112,
+            something2: index + 123456789101112,
+        };
+    });
+
+type RowType = typeof data[number];
+
+const columns: Column<RowType>[] = [
+    {
+        name: 'number',
+        width: 100,
+    },
+    {
+        name: 'col1',
+        width: 100,
+    },
+    {
+        name: 'col2',
+        width: 100,
+    },
+    {
+        name: 'string',
+        width: 100,
+        resizeMaxWidth: 200,
+    },
+    {
+        name: 'complex',
+        render: ({value}) => JSON.stringify(value, null, 2),
+        defaultOrder: DataTable.DESCENDING,
+        sortAscending: (item1, item2) => {
+            const value1 = item1.row.complex.value;
+            const value2 = item2.row.complex.value;
+            return (value1 % 2) - (value2 % 2) || value1 - value2;
+        },
+        width: 150,
+        resizeMinWidth: 100,
+    },
+    {
+        name: 'bar',
+        render: ({value}) => (
+            <div style={{width: value as number, height: 10, background: '#18b0ff'}} />
+        ),
+        resizeable: false,
+    },
+    {
+        name: 'multiLevel',
+        sub: [
+            {
+                name: 'sub1-1',
+                customStyle: () => ({backgroundColor: 'wheat', fontSize: '1.2em'}),
+                sub: [
+                    {
+                        name: 'sub2-1',
+                        resizeable: false,
+                    },
+                    {
+                        name: 'sub2-2',
+                        sub: [
+                            {
+                                name: 'sub3-1',
+                                accessor: 'something1',
+                                width: 100,
+                            },
+                        ],
+                    },
+                ],
+            },
+            {
+                name: 'something2',
+                width: 100,
+            },
+        ],
+    },
+];
+
+export function ResizeableTable({theme = YCLOUD_THEME, ...props}: DataTableProps<RowType>) {
+    const [tableColumnsWidthSetup, setTableColumnsWidth] = useTableResize('resizeableTable');
```

Comment:
does this approach will cause rerender whole table for each resize move? this may cause performance problems for big tables, maybe try to apply native resize solution, when resize affects only placeholder in header which affects whole column width?

Reply:
I didn't figured out how to resize only a single column in current realisation. It seems, everywhere it renders like `columns.map(renderFunction)`, so I cannot just update one column. And also width should be updated in many places at once (column, colgroup, rows, cells).

As for the native resize, it assumes defined resize icon, which I cannot change. Also native resize requires some css and html refactoring. And it will also cause whole table rerender, since we need update not only column, but also colgroup and cells (for proper content wrap) . So I applied solution, similar to default resize handler, that we have in internal common-datagrid.